File: Sources/windows/config/capabilities.py
from typing import Optional
from onscreen import GetsAngryWhenMeetingOtherCats, LeavesBreadcrumbs, LeavesPoopStains, RandomPlatformJumper, SleepingPlace
from yage.capabilities import *
from yage.models import *
from .sprites import MySpritesProvider

class MyCapabilities(CapabilitiesDiscoveryService):
    def __init__(self):
        self.capabilities = {
            "AnimatedSprite": AnimatedSprite,
            "AnimationsProvider": AnimationsProvider,
            "AnimationsScheduler": AnimationsScheduler,
            "AutoRespawn": AutoRespawn,
            "BounceOnLateralCollisions": BounceOnLateralCollisions,
            "Draggable": Draggable,
            "FlipHorizontallyWhenGoingLeft": FlipHorizontallyWhenGoingLeft,
            "GetsAngryWhenMeetingOtherCats": GetsAngryWhenMeetingOtherCats,
            "Gravity": Gravity,
            "LeavesBreadcrumbs": LeavesBreadcrumbs,
            "LeavesPoopStains": LeavesPoopStains,
            "LinearMovement": LinearMovement,
            "PetsSpritesProvider": MySpritesProvider,
            "RandomPlatformJumper": RandomPlatformJumper,
            "Rotating": Rotating,
            "Seeker": Seeker,
            "SleepingPlace": SleepingPlace,
            "WallCrawler": WallCrawler
        }

    def capability(self, id: str) -> Optional[Capability]:
        return self.capabilities.get(id)

File: Sources/macOS/Sources/OnScreen/Models/Capabilities.swift
import Foundation
import Yage

class PetsCapabilitiesDiscoveryService: CapabilitiesDiscoveryService {
    let capabilities: [String: Capability.Type] = [
        "AnimatedSprite": AnimatedSprite.self,
        "AnimationsProvider": AnimationsProvider.self,
        "AnimationsScheduler": AnimationsScheduler.self,
        "AutoRespawn": AutoRespawn.self,
        "BounceOnLateralCollisions": BounceOnLateralCollisions.self,
        "Draggable": Draggable.self,
        "FlipHorizontallyWhenGoingLeft": FlipHorizontallyWhenGoingLeft.self,
        "GetsAngryWhenMeetingOtherCats": GetsAngryWhenMeetingOtherCats.self,
        "Gravity": Gravity.self,
        "LeavesPoopStains": LeavesPoopStains.self,
        "LinearMovement": LinearMovement.self,
        "MouseChaser": MouseChaser.self,
        "PetsSpritesProvider": PetsSpritesProvider.self,
        "RandomPlatformJumper": RandomPlatformJumper.self,
        "Rotating": Rotating.self,
        "ShowsMenuOnRightClick": ShowsMenuOnRightClick.self,
        "SleepingPlace": SleepingPlace.self,
        "WallCrawler": WallCrawler.self
    ]
    
    func capability(for id: String) -> Capability? {
        capabilities[id]?.init()
    }
}
